internal/encryption/method/method.go

```diff
@@ -5,8 +5,18 @@
 
 package method
 
+// Config describes a configuration struct for setting up an encryption Method. You should always implement this
+// interface with a struct, and you should tag the fields with HCL tags so the encryption implementation can read
+// the .tf code into it. For example:
+//
+//	type MyConfig struct {
+//	    Key string `hcl:"key"`
+//	}
+//
+//	func (m MyConfig) Build() (Method, error) { ... }
 type Config interface {
 	// Build takes the configuration and builds an encryption method.
+	// TODO this may be better changed to return hcl.Diagnostics so warnings can be issued?
```

Comment:
+1

Reply:
Right now its not clear about where the boundary is between this package and opentofu code code.

When we pull out to an external package, I would like to strongly define a "boundary" where we say everything on one side is `error` and the other side is `hcl.Diagnostics` or `diags.Diagnostics`.

Reply:
Overall, we will leave this TODO in the code, and revisit later.